src/org/zaproxy/zap/extension/fuzz/httpfuzzer/processors/tagcreator/HttpFuzzerMessageProcessorTagUIPanel.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2015 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.extension.fuzz.httpfuzzer.processors.tagcreator;
+
+import org.parosproxy.paros.Constant;
+import org.zaproxy.zap.extension.fuzz.httpfuzzer.AbstractHttpFuzzerMessageProcessorUIPanel;
+
+import javax.swing.*;
+import java.awt.*;
+
+public class HttpFuzzerMessageProcessorTagUIPanel
+        extends AbstractHttpFuzzerMessageProcessorUIPanel<HttpFuzzerMessageProcessorTagCreator, HttpFuzzerMessageProcessorTagUI> {
+
+    private static final String MATCH_BY_REGEX_LABEL_REGEX = Constant.messages.getString("fuzz.httpfuzzer.processor.tagcreator.matchbyregex.regex");
+    private static final String MATCH_BY_REGEX_LABEL_TAG = Constant.messages.getString("fuzz.httpfuzzer.processor.tagcreator.matchbyregex.tag");
+    private static final String MATCH_BY_REGEX_NAME = Constant.messages.getString("fuzz.httpfuzzer.processor.tagcreator.matchbyregex.name");
+    private static final String EXTRACT_BY_REGEX_LABEL_REGEX = Constant.messages.getString("fuzz.httpfuzzer.processor.tagcreator.extractbyregex.regex");
+    private static final String EXTRACT_BY_REGEX_NAME = Constant.messages.getString("fuzz.httpfuzzer.processor.tagcreator.extractbyregex.name");
+    private final GridLayout layout;
+    private final JPanel fieldsPanel;
+    private ButtonGroup radioGroup;
+
+    private JTextField matchByRegExFieldRegEx;
```

Comment:
Consider using ZapTextField instead, it adds support for undo/redo (although it requires reset the undoes when closing the dialogue, and after setting the initial text).
Same comment for the other JTextField used.